app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -184,43 +156,7 @@ public Cursor getTalksByCenter(
             boolean isDownloaded
     )
     {
-        String query = "SELECT talks._id, talks.title, talks.teacher_id, centers.name, talks.rec_date FROM talks";
-
-        if (isStarred)
-        {
-            query += joinStarredTalks();
-        }
-
-        if (isDownloaded)
-        {
-            query += joinDownloadedTalks();
-        }
-
-        query += joinTalkIdVenueId();
-
-        String where = " WHERE ";
-
-        if (searchTerms != null && !searchTerms.isEmpty())
-        {
-            String[] selectionColumns = new String[]
-            {
-                DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.TITLE,
-                DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.DESCRIPTION,
-            };
-
-            where += getSearchStatement(searchTerms, selectionColumns) + " AND ";
-        }
-
-        where += String.format(
-                " %s.%s = %s ",
-                DBManager.C.Talk.TABLE_NAME,
-                DBManager.C.Talk.VENUE_ID,
-                venueId
-        );
-
-        query += where + " ORDER BY talks.rec_date DESC ";
-
-        return queryIfNotNull(query, null);
+        return getTalks(searchTerms, isStarred, isDownloaded, DBManager.C.Center.TABLE_NAME + "." + DBManager.C.Center.ID + "=" + venueId);
```

Comment:
thanks for not being lazy like me and using the constant table names instead of hardcoding them

Reply:
Haha, I actually wanted to ask you whether you felt that having all of these constants is worth it. :P  I'm kind of on the fence myself.  It's nice to have the extra assurance that you haven't introduced a typo in a field name, but it arguably also makes the code harder to read and write, especially since it forces you to do lots of verbose string concatenation all over the place.  It'd be probably be good to standardize on one way of doing it, but I'm totally open to opinions on which way to go.